autoload/go/def.vim

```diff
@@ -84,17 +93,29 @@ function! s:godefJump(out, mode)
 	let old_errorformat = &errorformat
 	let &errorformat = "%f:%l:%c"
 
-	if a:out =~ 'godef: '
-		let out = substitute(a:out, go#util#LineEnding() . '$', '', '')
-		echom out
+	" Location is the first line of godef output. Ideally in the proper format
+	" but it could also be an error
+	let location = a:out[0]
+
+	" Echo the godef error if we had one.
+	if location =~ 'godef: '
+		let gderr=substitute(location, go#util#LineEnding() . '$', '', '')
+		echom gderr
+		" Don't jump if we're in a modified buffer
+	elseif getbufvar(bufnr('%'), "&mod")
+        echohl ErrorMsg
+		echo "No write since last change"
+        echohl None
```

Comment:
I've fixed the error, and I'll exchange the echom for the util version as well. Returning is unnecessary, though, unless I've made a huge mistake, as all of the work is done in the else clause.  If you'd like it to be more go-y, I'd be happy to do early returns.  I'd probably put the modified check first in that case.